lib/db.js

```diff
@@ -0,0 +1,135 @@
+import sqlite3 from 'sqlite3'
+import joi from 'joi'
+
+// The constants below can be passed to initDB to create a persistent or in-memory database.
+export const DB_PERSISTENT = "re-employment-kraken.db"
+export const DB_IN_MEMORY = ":memory:"
+
+// Due to simplicity, all schema definition steps are defined below. Depending on required future changes,
+// we can think about introducing a schema migration tool and extract all steps to plain SQL files.
+// Please make sure that schema changes are backwards-compatible and idempotent:
+// 1. Changes should not break existing databases.
+// 2. Running those steps repeatedly should not change the schema and not cause any errors or data loss.
+const CREATE_SCHEMA = `
+        CREATE TABLE IF NOT EXISTS job (
+            id INTEGER PRIMARY KEY AUTOINCREMENT, -- Use AUTOINCREMENT to prevent reuse of deleted rows ROWID.
+            strategy_name TEXT,
+            query TEXT,
+            title TEXT,
+            url TEXT NOT NULL UNIQUE -- The url key allows deduplication and detection of known projects.
```

Comment:
I basically copied the table definition from my own project with small adaptions. Coming from PostgreSQL and Java, I usually use lower snake case for SQL code (since it is case-insensitive).
However, I get your point. For a language like JavaScript it makes sense to use the exact same naming to simplify our mapping.

Reply:
Ah, that makes sense. Sorry, I didn't think about SQL conventions. A proper mapping would be fine too, of course.